driver-async/src/main/com/mongodb/async/client/OperationExecutorImpl.java

```diff
@@ -122,6 +122,9 @@ private AsyncReadWriteBinding getReadWriteBinding(final ReadPreference readPrefe
             if (session.hasActiveTransaction() && !readPreference.equals(primary())) {
                 throw new MongoClientException("Read preference in a transaction must be primary");
             }
+            if (!session.hasActiveTransaction() && session.getOptions().getAutoStartTransaction()) {
```

Comment:
@rstam this seemed clean enough to do here and would have required more code changes to do it way down in CommandMessage, as you essentially did in the .NET driver.  The price is that I had to do it twice, once for async and once for sync.

Reply:
Ack